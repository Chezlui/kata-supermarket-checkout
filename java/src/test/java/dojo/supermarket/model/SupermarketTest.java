package dojo.supermarket.model;

import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SupermarketTest {
    private SupermarketCatalog catalog;
    private Teller teller;
    private ShoppingCart theCart;
    private Product toothbrush;
    private Product rice;
    private Product apples;

    @Before
    public void setUp() {
        catalog = new DummyCatalog();
        teller = new Teller(catalog);
        theCart = new ShoppingCart();

        toothbrush = new Product("toothbrush", ProductUnit.Each);
        catalog.addProduct(toothbrush, 0.99);
        rice = new Product("rice", ProductUnit.Each);
        catalog.addProduct(rice, 2.99);
        apples = new Product("apples", ProductUnit.Kilo);
        catalog.addProduct(apples, 1.99);

    }

    @Test
    public void an_empty_shopping_cart_should_cost_nothing() {
        Receipt receipt = teller.checksOutArticlesFrom(theCart);
        Approvals.verify(ReceiptPresenter.present(receipt));
    }

    @Test
    public void one_normal_item() {
        theCart.addItem(toothbrush);
        Receipt receipt = teller.checksOutArticlesFrom(theCart);
        Approvals.verify(ReceiptPresenter.present(receipt));
    }

    @Test
    public void two_normal_items() {
        theCart.addItem(toothbrush);
        theCart.addItem(rice);
        Receipt receipt = teller.checksOutArticlesFrom(theCart);
        Approvals.verify(ReceiptPresenter.present(receipt));
    }

    @Test
    public void buy_two_get_one_free() {
        theCart.addItem(toothbrush);
        theCart.addItem(toothbrush);
        theCart.addItem(toothbrush);
        teller.addSpecialOffer(new ThreeForThePriceOfTwo(toothbrush, catalog.getUnitPrice(toothbrush)));
        Receipt receipt = teller.checksOutArticlesFrom(theCart);
        Approvals.verify(ReceiptPresenter.present(receipt));
    }

    @Test
    public void loose_weight_product() {
        theCart.addItemQuantity(apples, .5);
        Receipt receipt = teller.checksOutArticlesFrom(theCart);
        Approvals.verify(ReceiptPresenter.present(receipt));
    }

}
