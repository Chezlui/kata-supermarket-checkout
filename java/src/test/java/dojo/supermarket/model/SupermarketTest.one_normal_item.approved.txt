ReceiptItem{product=toothbrush, quantity=1, price=0.99, totalPrice=0.99}
Total: 0.99